workers/unity/Packages/com.improbable.gdk.core/Systems/ComponentUpdateSystem.cs

```diff
@@ -21,62 +20,62 @@ public class ComponentUpdateSystem : ComponentSystem
 
         public void SendUpdate<T>(T update, EntityId entityId) where T : ISpatialComponentUpdate
         {
-            messagesToSend.AddComponentUpdate(update, entityId.Id);
+            worker.MessagesToSend.AddComponentUpdate(update, entityId.Id);
         }
 
         public void SendEvent<T>(T eventToSend, EntityId entityId) where T : IEvent
         {
-            messagesToSend.AddEvent(eventToSend, entityId.Id);
+            worker.MessagesToSend.AddEvent(eventToSend, entityId.Id);
         }
 
         public ReceivedMessagesSpan<ComponentEventReceived<T>> GetEventsReceived<T>() where T : IEvent
         {
-            var manager = (IDiffEventStorage<T>) viewDiff.GetComponentDiffStorage(typeof(T));
+            var manager = (IDiffEventStorage<T>) worker.Diff.GetComponentDiffStorage(typeof(T));
```

Comment:
Can cache the diff?

Reply:
And `MessagesToSend` above

Reply:
We can but there is a question of when. As the diff doesn't have to be the same ref, it just happens to be for now.